build_tools/benchmarks/collect_compilation_statistics.py

```diff
@@ -130,6 +133,81 @@ def get_module_path(flag_file: TextIO) -> Optional[str]:
   return module_path
 
 
+def get_module_map_from_generation_config(
+    serialized_gen_config: TextIO, e2e_test_artifacts_dir: pathlib.PurePath
+) -> Dict[CompilationInfo, pathlib.Path]:
+  gen_configs = serialization.unpack_and_deserialize(
+      data=json.load(serialized_gen_config),
+      root_type=List[iree_definitions.ModuleGenerationConfig])
+  module_map = {}
+  for gen_config in gen_configs:
+    model = gen_config.imported_model.model
+    compile_config = gen_config.compile_config
+    target_archs = []
+    for compile_target in compile_config.compile_targets:
+      arch = compile_target.target_architecture
+      target_archs.append(
+          (f"{arch.type.value}-{arch.architecture}-{arch.microarchitecture}-"
+           f"{compile_target.target_abi.value}"))
+    compilation_info = CompilationInfo(
+        model_name=model.name,
+        model_tags=tuple(model.tags),
```

Comment:
There are many dataclasses' fields needed to replace List with Tuple, I prefer not to touch that many lines here.
Filed an issue to track the refactor #11588